src/core/server/http/router/request.ts

```diff
@@ -91,17 +101,15 @@ export class KibanaRequest<
   public static from<P, Q, B>(
     req: Request,
     routeSchemas: RouteValidator<P, Q, B> | RouteValidatorFullConfig<P, Q, B> = {},
-    withoutSecretHeaders: boolean = true
+    { withoutSecretHeaders = true, requestOpaqueId = {} }: DeepPartialObject<RequestOptions> = {}
   ) {
     const routeValidator = RouteValidator.from<P, Q, B>(routeSchemas);
     const requestParts = KibanaRequest.validate(req, routeValidator);
-    return new KibanaRequest(
-      req,
-      requestParts.params,
-      requestParts.query,
-      requestParts.body,
-      withoutSecretHeaders
-    );
+    return new KibanaRequest(req, requestParts.params, requestParts.query, requestParts.body, {
+      withoutSecretHeaders,
+      // Use the config schema to populate defaults
+      requestOpaqueId: requestIdConfigSchema.validate(requestOpaqueId),
```

Comment:
Hasn't it been validated in the config?

Reply:
I was using this primarily for populating the defaults, but in light of your comment https://github.com/elastic/kibana/pull/71019#discussion_r451539761, this will need to move.

Reply:
Removed